### PR TITLE
Build primarily with `dotnet msbuild`

### PR DESCRIPTION
**Pull request by dougbu**
- make `dotnet msbuild` the default on Windows too
- enable building managed projects depending on native assets
- revert move to VS2019.Pre queues

### CODE REVIEW

**Comment by dougbu** [comment]
See https://dev.azure.com/dnceng/internal/_build/results?buildId=662159 for internal validation of this change.

File: eng/Version.Details.xml
Comment (dougbu) [inline eng/Version.Details.xml:291]: Ignore everything from the Arcade update commit [fc607e97fc3c](https://github.com/dotnet/aspnetcore/pull/22017/commits/fc607e97fc3c2e9bd1158ea4b7b56bbab8c73515). Arcade should be promoting a build containing [fc3b1e63de02](https://github.com/dotnet/arcade/commit/fc3b1e63de02e2ab97e638959728c9a98dfbac11) real soon now.

File: eng/scripts/GenerateProjectList.ps1
Comment (dougbu) [inline eng/scripts/GenerateProjectList.ps1:17]: Copied this stuff from our /build.ps1 file because invoking tools.ps1 corrupts global state if this script, eng/scripts/GenerateReferenceAssemblies.ps1, or eng/scripts/CodeCheck.ps1 are invoked in an interactive PowerShell session.

File: eng/scripts/GenerateReferenceAssemblies.ps1
Comment (dougbu) [inline eng/scripts/GenerateReferenceAssemblies.ps1:10]: Building native assets to avoid warnings when running the `GenerateReferenceSources` target.

File: src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj
Comment (dougbu) [inline src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj:144]: Unlike other managed projects with references to native projects, this one didn't have a fallback to find the right files and pack them up.

File: build.ps1
Comment (dougbu) [inline build.ps1:378]: Not relevant in CI but will be helpful when we re-enable binary logs.

File: .azure/pipelines/helix-matrix.yml
Comment (HaoK) [inline .azure/pipelines/helix-matrix.yml:32]: Nit: is this comment correct still?  Looks like we are just building grpc now with this step?

Comment (dougbu) [reply .azure/pipelines/helix-matrix.yml:32]: This is now correct. The comment was wrong before because it would restore the default `--all` projects, plus the gRPC (Interop) projects.

**Comment by dougbu** [comment]
/ping reviewers (between trainings of course): Latest build (after merging changes from master) was successful.

File: .azure/pipelines/ci.yml
Comment (wtgodbe) [reply .azure/pipelines/ci.yml:308]: Even if it's a no-op, I think it's more clear to keep it so that it's obvious to devs what arch is being built

File: Directory.Build.props
Comment (wtgodbe) [inline Directory.Build.props:183]: What's the motivation for this logical change?

File: eng/scripts/GenerateReferenceAssemblies.ps1
Comment (wtgodbe) [reply eng/scripts/GenerateReferenceAssemblies.ps1:10]: How long does it take to build native assets?

**Review by wtgodbe — APPROVED** [review]
Looks good outside of the few questions

File: eng/AfterSolutionBuild.targets
Comment (JunTaoLuo) [inline eng/AfterSolutionBuild.targets:9]: why was the condition added?

**Review by JunTaoLuo — APPROVED** [review]

File: eng/scripts/GenerateReferenceAssemblies.ps1
Comment (dougbu) [reply eng/scripts/GenerateReferenceAssemblies.ps1:10]: Looking at hhttps://dev.azure.com/dnceng/internal/_build/results?buildId=662159&view=logs&j=1b89928a-2219-5ef9-602f-f95beb3da4dc&t=bcce746a-741e-59b7-adec-a4a82d8c5e52&l=313 for example, 7 minutes and 40 seconds. I believe this aligns with a speedup in the next `MSBuild` invocation.

The overall build time goes down across all jobs comparing builds of this PR (2h 3m 45s average of 5 builds) with 'master' (2h 30m 45s average of 17 builds in the last 7 days). Performance wasn't the goal but it still seemed to improve. That said, 5 builds isn't that many.

Part of the improvement may come from only needing to build native assets once per build job. `-noBuildNative` was not possible everywhere before because the native assets were built only for a single architecture.

File: Directory.Build.props
Comment (dougbu) [reply Directory.Build.props:183]: The `$(BuildIisNativeProjects)` property is really badly named. It actually controls whether the native assets are even looked for. We had the infrastructure in place to find the assets if they were build earlier but weren't using it. So, without this change, the native build would produce assemblies the managed build ignored.

File: eng/AfterSolutionBuild.targets
Comment (dougbu) [reply eng/AfterSolutionBuild.targets:9]: A single `build.cmd` invocation now runs the `MSBuild` PowerShell function twice (three times including the repo tasks) and both would redo this step if the dev had built the repo earlier. Since the fixup only matters for managed packages and we only build managed packages in the `dotnet msbuild` invocation, this is safe and quiets things down.

**Comment by dougbu** [comment]
:up::date: to rebase on 'master', undo `-arch x64` removals, and add a few more `-nobl` and `-noBuildRepoTools` command-line arguments.

**Comment by dougbu** [comment]
:ok: looking good for PR validation and https://dev.azure.com/dnceng/internal/_build/results?buildId=664525 built the commit right before the fix-ups of scripts that aren't exercised in internal builds.

**Pull request merged**
